records/generator.go

```diff
@@ -21,12 +23,15 @@ import (
 // Will likely become map[string][]discoveryinfo
 type rrs map[string][]string
 
+var defaultDomainPatterns = []string{"{{.tname}}"}
+
 // Mesos-DNS state
 // Refactor when discovery id is available
 type RecordGenerator struct {
 	As     rrs
 	SRVs   rrs
 	Slaves map[string]string
+	DomainPatterns []string
```

Comment:
I have moved the pattern list out of the RecordGenerator again. Now it's a parameter to `InsertState` like all other settings coming from the config.